python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_cli_event.py

```diff
@@ -332,15 +332,24 @@ def to_default_asset_events(
             "invocation_id": invocation_id,
         }
 
-        if event_node_info.get("node_started_at") and event_node_info.get("node_finished_at"):
+        if event_node_info.get("node_started_at") == "" and event_node_info.get("node_finished_at") == "":
+            # if model materialization is incremental microbatch, node_started_at and node_finished_at are empty strings
+            # and require fallback to data.execution_time
+            default_metadata["Execution Duration"] = self.raw_event["data"]["execution_time"]
+        elif event_node_info.get("node_started_at") and event_node_info.get("node_finished_at"):
             started_at = dateutil.parser.isoparse(event_node_info["node_started_at"])
             finished_at = dateutil.parser.isoparse(event_node_info["node_finished_at"])
             default_metadata["Execution Duration"] = (finished_at - started_at).total_seconds()
 
         has_asset_def: bool = bool(context and context.has_assets_def)
 
         node_resource_type: str = event_node_info["resource_type"]
-        node_status: str = event_node_info["node_status"]
+        # if model materialization is incremental microbatch, node_status property is "None", hence fall back to status
+        node_status: str = (
+            self.raw_event["data"]["status"].lower()
+            if event_node_info["node_status"] == "None"
```

Comment:
Same question here

Reply:
see above. I wasn't able to find a prove in the code that it would be consistently `"None"` or `""` or `None`